### PR TITLE
feat(dew): add a new datasource to get list of grants

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new datasource to get list of grants.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new datasource to get the list of the grants.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDataSourceKmsGrants_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDataSourceKmsGrants_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceKmsGrants_basic
=== PAUSE TestAccDataSourceKmsGrants_basic
=== CONT  TestAccDataSourceKmsGrants_basic
--- PASS: TestAccDataSourceKmsGrants_basic (54.55s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       54.603s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/data-sources/kms_grants.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+subcategory: "Data Encryption Workshop (DEW)"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_kms_grants"
+description: |-
+  Use this datasource to get list of the grants.
+---
+
+# huaweicloud_kms_grants
+
+Use this datasource to get list of the grants.
```
Comment (deer-hang) [inline docs/data-sources/kms_grants.md]:
```suggestion
Use this datasource to get a list of grants.
```
same error to line 6

File: huaweicloud/services/acceptance/dew/data_source_huaweicloud_kms_grants_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+					resource.TestCheckResourceAttrSet(dataSource, "grants.0.id"),
+					resource.TestCheckResourceAttrSet(dataSource, "grants.0.key_id"),
+					resource.TestCheckResourceAttrSet(dataSource, "grants.0.type"),
+					resource.TestCheckResourceAttrSet(dataSource, "grants.0.creator"),
+					resource.TestCheckResourceAttrSet(dataSource, "grants.0.grantee_principal"),
+					resource.TestCheckResourceAttrSet(dataSource, "grants.0.operations.#"),
+					resource.TestCheckResourceAttrSet(dataSource, "grants.0.created_at"),
+
+					dc.CheckResourceExists(),
+					resource.TestCheckOutput("is_exist_grant", "true"),
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/dew/data_source_huaweicloud_kms_grants_test.go]: It is no need to check whether the grants is existed.
Remove the repeat `dc.CheckResourceExists(),`

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccDataSourceKmsGrants_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDataSourceKmsGrants_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceKmsGrants_basic
=== PAUSE TestAccDataSourceKmsGrants_basic
=== CONT  TestAccDataSourceKmsGrants_basic
--- PASS: TestAccDataSourceKmsGrants_basic (52.63s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       52.679s
```

**Pull request merged**